components/virtual-node/virtual-node-rest-service-impl/src/main/kotlin/net/corda/virtualnode/rest/impl/v1/VirtualNodeRestResourceImpl.kt

```diff
@@ -367,20 +370,65 @@ internal class VirtualNodeRestResourceImpl(
         }
     }
 
-    override fun getCreateCryptoSchemaSQL(): ResponseEntity<String> {
-        TODO("To be implemented in CORE-15805")
+    override fun getCreateCryptoSchemaSQL(): String {
+        return getSchemaSql(DbTypes.CRYPTO, null, null)
     }
 
-    override fun getCreateUniquenessSchemaSQL(): ResponseEntity<String> {
-        TODO("To be implemented in CORE-15805")
+    override fun getCreateUniquenessSchemaSQL(): String {
+        return getSchemaSql(DbTypes.UNIQUENESS, null, null)
     }
 
-    override fun getCreateVaultSchemaSQL(cpiChecksum: String): ResponseEntity<String> {
-        TODO("To be implemented in CORE-15805")
+    override fun getCreateVaultSchemaSQL(cpiChecksum: String): String {
+        return getSchemaSql(DbTypes.VAULT, null, cpiChecksum)
     }
 
-    override fun getUpdateSchemaSQL(virtualNodeShortId: String, newCpiChecksum: String): ResponseEntity<String> {
-        TODO("To be implemented in CORE-15805")
+    override fun getUpdateSchemaSQL(virtualNodeShortId: String, newCpiChecksum: String): String {
+        return getSchemaSql(DbTypes.VAULT, virtualNodeShortId, newCpiChecksum)
+    }
+
+    private fun getSchemaSql(
+        dbType: DbTypes,
+        virtualNodeShortId: String?,
+        cpiChecksum: String?
+    ): String {
+        val instant = clock.instant()
+
+        val rpcRequest = VirtualNodeManagementRequest(
+            instant,
+            VirtualNodeSchemaRequest(
+                dbType,
+                virtualNodeShortId,
+                cpiChecksum
+            )
+        )
+
+        val operationLog = when (dbType) {
+            DbTypes.CRYPTO -> "get Schema SQL to create Crypto DB"
+            DbTypes.UNIQUENESS -> "get Schema SQL to create Uniqueness DB"
+            DbTypes.VAULT -> {
+                if (virtualNodeShortId.isNullOrBlank()) {
+                    "get Schema SQL to create Vault DB and CPI"
+                } else {
+                    "get Schema SQL to update CPI"
+                }
+            }
+        }
+
+        // Send request and await response message on bus
+        val resp = tryWithExceptionHandling(logger, operationLog) {
+            sendAndReceive(rpcRequest)
+        }
+
+        return when (val resolvedResponse = resp.responseType) {
+            is VirtualNodeSchemaResponse -> {
+                resolvedResponse.run {
+                    schemaSql
+                }
```

Comment:
```suggestion
                resolvedResponse.schemaSql
```
... it is just shorter.